get-digest.sh

```diff
@@ -1,38 +1,25 @@
 #!/bin/bash
 
-if [ "$#" -ne 2 ]
+if [ "$#" -ne 3 ]
 then
-    echo "Need ARO_HCP_IMAGE_ACR and REPOSITORY parameters"
+    echo "Usage: $0 <ARO_HCP_IMAGE_ACR> <REPOSITORY> <TAG>"
     exit 1
 fi
 
 aro_hcp_image_acr=${1}
 repository=${2}
+tag=${3}
 
-if [ -n "${IMAGE_DIGEST_OVERRIDE}" ];
-then
-    echo ${IMAGE_DIGEST_OVERRIDE}
-    exit 0
-fi
-
-if [ -n "${IMAGE_DIGEST}" ];
-then
-    echo ${IMAGE_DIGEST}
-    exit 0
-fi
-
-
-tags=$(mktemp)
-trap "rm ${tags}" EXIT
-
-az acr repository show-tags --orderby time_desc --n ${aro_hcp_image_acr} --repository ${repository} --detail > $tags
+tags=$(az acr repository show-tags --orderby time_desc --n "${aro_hcp_image_acr}" --repository "${repository}" --detail)
 
-suggested_digest=$(jq -r --arg TAG $(git rev-parse --short=7 HEAD) \
-    'first(.[] | select(.name==$TAG) | .digest)' $tags)
-if [ -n "${suggested_digest}" ];
+# find the digest for the specified tag
+suggested_digest=$(jq -r --arg TAG "${tag}" \
+    'first(.[] | select(.name==$TAG) | .digest)' <<< "${tags}")
+if [ -n "${suggested_digest}" ] && [ "${suggested_digest}" != "null" ];
 then
-    echo ${suggested_digest}
+    echo "${suggested_digest}"
     exit 0
 fi
 
-jq -r 'first(.[] | .digest)' $tags
\ No newline at end of file
+# if there is no digest for the specified tag, return the first digest from the list of non-dirty tags
+jq -r 'first(.[] | select(.name | endswith("-dirty") | not) | .digest)' <<< "${tags}"
```

Comment:
i'm not sure we should actually keep this
* non-RP devs will never hit this situation as they rely on the digests from the config
* RP devs will build and push the image as part of `make deploy.current` anyways